frontend/src/components/navbar/NavButton.tsx

```diff
@@ -0,0 +1,21 @@
+import './NavButton.scss'
+import {ReactElement} from "react";
+
+interface NavButtonProps<T, TResult> {
+    icon?: ReactElement
+    currentPage?: string
+    value?: string
+    onClick?: (item: T) => TResult
+}
+
+export default function NavButton({icon, value, onClick, currentPage}: NavButtonProps<any, any>) {
+
+    const pageClass: string = currentPage === value ? " highlight" : ""
```

Comment:
I would have a look at the react router hooks, for an alternative that saaving the currentPage.